nbgitpuller/static/js/gitsyncview.js

```diff
@@ -58,8 +59,10 @@ export class GitSyncView{
     setProgressError(isError) {
         if (isError) {
             this.progress.classList.add('progress-bar-danger');
+            this.recovery.classList.remove('hidden');
         } else {
             this.progress.classList.remove('progress-bar-danger');
+            this.recovery.classList.add('hidden');
         }
     }
```

Comment:
```suggestion
    setProgressError(isError) {
        if (isError) {
            this.progress.classList.add('progress-bar-danger');
        } else {
            this.progress.classList.remove('progress-bar-danger');
        }
    }

    setRecoveryLink(isError) {
        if (isError) {
            this.recovery.classList.toggle('hidden', !visible);
        }
    }
```

Separating this into its own class method will help with clarifying and reasoning about its functionality in the future, rather than burying it inside the existing method for changing the progress bar class list.